application/src/org/commcare/util/J2METransactionParserFactory.java

```diff
@@ -63,15 +65,13 @@ public CommCareTransactionParserFactory(boolean tolerant) {
     public TransactionParser getParser(KXmlParser parser) {
         String namespace = parser.getNamespace();
         String name = parser.getName();
-        if ("case".equalsIgnoreCase(name)) {
-            return new AttachableCaseXMLParser(parser, caseTallies, tolerant, (IStorageUtilityIndexed)StorageManager.getStorage(Case.STORAGE_KEY));
-        } else if("registration".equalsIgnoreCase(name)) {
-            //TODO: It's possible we want to do the restoreID thing after signalling success, actually. If the
-            //restore gets cut off, we don't want to be re-sending the token, since it implies that it worked.
-            return new UserXmlParser(parser, restoreId);
-        }  else if(LedgerXmlParsers.STOCK_XML_NAMESPACE.equalsIgnoreCase(namespace)) {
-            return new LedgerXmlParsers(parser, (IStorageUtilityIndexed)StorageManager.getStorage(Ledger.STORAGE_KEY));
-        } else if("message".equalsIgnoreCase(name)) {
+
+        TransactionParser superParser = super.getParser(parser);
+        if(superParser != null){
```

Comment:
I wonder if for future-proofing this check should come _after_ all of the internal ones? Seems like we might want to default to the assumption that we override the superclass's parsing behavior if we supply our own parser for something.